src/JustEat.StatsD/SocketTransport.cs

```diff
@@ -29,6 +29,11 @@ public sealed class SocketTransport : IStatsDTransport, IStatsDBufferedTransport
         public SocketTransport(IPEndPointSource endPointSource, SocketProtocol socketProtocol)
         {
             _endpointSource = endPointSource ?? throw new ArgumentNullException(nameof(endPointSource));
+
+            if (socketProtocol != SocketProtocol.IP && socketProtocol != SocketProtocol.Udp)
```

Comment:
If you use [`Enum.IsDefined(Type, object)`](https://docs.microsoft.com/en-gb/dotnet/api/system.enum.isdefined?view=netframework-4.7.2#System_Enum_IsDefined_System_Type_System_Object_) then we don't need to change this code if we add a new value to the enum.